presentation/src/main/java/com/open/presentation/ui/WeatherScreen.kt

```diff
@@ -0,0 +1,89 @@
+package com.open.presentation.ui
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.open.data.network.UiState
+import com.open.presentation.ui.viewmodel.WeatherViewModel
+
+@Composable
+fun WeatherScreen(
+    viewModel: WeatherViewModel = hiltViewModel()
+) {
+    var city by remember { mutableStateOf("") }
+    val weatherState by viewModel.weatherState.collectAsState()
+
+    // Debugging Column setup
+    Column(
+        modifier = Modifier,
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        // Debug: Ensure BasicTextField is rendered
```

Comment:
Make this a TODO if you want to remember it, otherwise it's not providing any value here.